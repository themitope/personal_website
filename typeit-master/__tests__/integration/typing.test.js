import TypeIt from "../../src/TypeIt";

test("Generates a queue correctly.", () => {
  setHTML`<div>'
      <span id="element"></span>
    </div>`;

  const instance = new TypeIt("#element", {
    strings: ["Taxation is...", "theft."]
  }).go();

  expect(instance.getQueue().getItems()).toMatchSnapshot();
});

test("Generates a queue correctly when chaining upon instantiation.", () => {
  setHTML`<div>'
      <span id="element"></span>
    </div>`;

  const instance = new TypeIt("#element", {})
    .type("First string.")
    .delete()
    .type("Second string.")
    .go();

  expect(instance.getQueue().getItems()).toHaveLength(35);
});

test("Generates correct `nextStringDelay`.", () => {
  setHTML`<div>'
      <span id="element"></span>
    </div>`;

  const instance1 = new TypeIt("#element", {
    nextStringDelay: 500,
    strings: ["Free markets...", "win."]
  }).go();

  let nextStringDelay = instance1.getOptions().nextStringDelay;

  expect(typeof nextStringDelay).toBe("object");
  expect(nextStringDelay.before).toBe(250);
  expect(nextStringDelay.after).toBe(250);
  expect(nextStringDelay.total).toBe(500);

  const instance2 = new TypeIt("#element", {
    nextStringDelay: [150, 400],
    strings: ["Free markets...", "win."]
  }).go();

  nextStringDelay = instance2.getOptions().nextStringDelay;

  expect(nextStringDelay.before).toBe(150);
  expect(nextStringDelay.after).toBe(400);
  expect(nextStringDelay.total).toBe(550);
});

test("Generates correct `loopDelay`.", () => {
  setHTML`<div>'
      <span id="element"></span>
    </div>`;

  const instance1 = new TypeIt("#element", {
    nextStringDelay: 500,
    strings: ["Free markets...", "win."]
  }).go();

  let loopDelay = instance1.getOptions().loopDelay;

  expect(loopDelay.before).toBe(375);
  expect(loopDelay.after).toBe(375);
  expect(loopDelay.total).toBe(750);

  const instance2 = new TypeIt("#element", {
    loopDelay: [3000, 5000],
    strings: ["Free markets...", "win."]
  }).go();

  loopDelay = instance2.getOptions().loopDelay;

  expect(typeof loopDelay).toBe("object");
  expect(loopDelay.before).toBe(3000);
  expect(loopDelay.after).toBe(5000);
  expect(loopDelay.total).toBe(8000);
});

test("Removes empty HTML when necessary.", () => {
  setHTML`<div>'
      <span id="element"></span>
    </div>`;

  const instance = new TypeIt("#element", {
    breakLines: false
  })
    .type("This is a string with some <strong>bold.</strong>")
    .delete(5)
    .type("standard text.")
    .delete(14)
    .type("<em>italicized text...</em>")
    .delete(18)
    .type("standard text again.")
    .go();

  let emptyTagPattern = /<[^\/>][^>]*><\/[^>]+>/;

  let result = document.getElementById("element").innerHTML;

  //-- Ensure our regex correctly finds empty tags.
  expect("Text with <strong></strong> a set of empty tags!").toMatch(
    emptyTagPattern
  );

  //-- Will not have any empty HTML.
  expect(result).not.toMatch(emptyTagPattern);
});
