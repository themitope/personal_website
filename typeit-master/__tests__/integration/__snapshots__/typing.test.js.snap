// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates a queue correctly. 1`] = `
Array [
  Array [
    [Function],
    250,
    Object {
      "id": 0,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "T",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 1,
      "isFirst": true,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "a",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 2,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "x",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 3,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "a",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 4,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "t",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 5,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "i",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 6,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "o",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 7,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "n",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 8,
    },
  ],
  Array [
    [Function],
    Object {
      "content": " ",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 9,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "i",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 10,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "s",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 11,
    },
  ],
  Array [
    [Function],
    Object {
      "content": ".",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 12,
    },
  ],
  Array [
    [Function],
    Object {
      "content": ".",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 13,
    },
  ],
  Array [
    [Function],
    Object {
      "content": ".",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": Taxation is...,
    },
    Object {
      "freezeCursor": true,
      "id": 14,
      "isLast": true,
    },
  ],
  Array [
    [Function],
    375,
    Object {},
  ],
  Array [
    [Function],
    Object {
      "content": <br />,
      "isHTMLElement": true,
      "isTopLevelText": false,
      "node": null,
    },
    Object {
      "freezeCursor": true,
      "id": 15,
    },
  ],
  Array [
    [Function],
    375,
    Object {},
  ],
  Array [
    [Function],
    Object {
      "content": "t",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": theft.,
    },
    Object {
      "freezeCursor": true,
      "id": 16,
      "isFirst": true,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "h",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": theft.,
    },
    Object {
      "freezeCursor": true,
      "id": 17,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "e",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": theft.,
    },
    Object {
      "freezeCursor": true,
      "id": 18,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "f",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": theft.,
    },
    Object {
      "freezeCursor": true,
      "id": 19,
    },
  ],
  Array [
    [Function],
    Object {
      "content": "t",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": theft.,
    },
    Object {
      "freezeCursor": true,
      "id": 20,
    },
  ],
  Array [
    [Function],
    Object {
      "content": ".",
      "isHTMLElement": false,
      "isTopLevelText": true,
      "node": theft.,
    },
    Object {
      "freezeCursor": true,
      "id": 21,
      "isLast": true,
    },
  ],
]
`;
